### DISCUSSION TITLE
Giu how to set drag, zoom, close, zoom in and out window events in g.MasterWindowFlagsFramel?

### DISCUSSION THREAD

**Discussion by Seikaijyu** [AllenDang/giu#538] [Q&A] [answered]
I want to make a custom form style application, how to let other controls inherit form events?

**Comment by gucio321** [comment, marked as answer]
Imo it is something related to glfw's stuff.  i don't know it's code as well as gou's. But i suppose you need to search there: https://github.com/go-gl/glfw

**Reply by gucio321** [reply]
Yeah glfw.Window has MaximalizeCallback method so now you just need to get glfw.Window some way... @AllenDang  how to extract it from imgui-go?

**Reply by AllenDang** [reply]
I don't quite understand, what are we discussing here?

**Reply by gucio321** [reply]
If i understand correctly it's just "how to dry close/ maximalize/ hide callbacks for master window" however i can be wrong

**Reply by AllenDang** [reply]
@SuperLiveYou Can you tell us more detail about what are you trying to do?

**Reply by Seikaijyu** [reply]
Well, I hope to customize the MasterWindow style and implement related events, such as dragging the form event, zooming in and out event, etc
@AllenDang

**Reply by Seikaijyu** [reply]
Maybe my description is not very clear

**Reply by Seikaijyu** [reply]
This is a demonstration
https://user-images.githubusercontent.com/68497295/180592769-33fbc4f7-4586-4e34-aa7a-1b6eb33c4c92.mp4
@AllenDang

**Reply by AllenDang** [reply]
You mean customize the styoe of MasterWindow? This could be done by create a frameless windows and set style of a g.Window.
And for the events you described, giu doesn't have them now, but I believe GLFW has some, not sure about zooming in and out, it's easy to expose them into giu.
But I still need to know more about what you are trying to archive. To expose an API is easy, but to expose them right with a well organized interface is not.

**Reply by Seikaijyu** [reply]
OK, thank you very much for your answer
